D/D.sublime-syntax

```diff
@@ -8,339 +8,2057 @@ file_extensions:
+        - match: \bunittest\b
+          scope: keyword.control.conditional.d
+          set: conditional-declaration-after-parens
+        - match: \bassert\b
+          scope: keyword.other.assert.d
+          set: conditional-declaration-after-parens
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: conditional-declaration-after-parens
+        - match: '{{integer_lookahead}}'
+          set: [conditional-declaration-after-parens, integer]
+        - include: not-whitespace-illegal
+    - match: '='
+      scope: keyword.operator.assignment.d
+      set: conditional-declaration-assignment
+    - include: not-whitespace-illegal
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: conditional-declaration-after-parens
+        - match: '{{integer_lookahead}}'
+          set: [conditional-declaration-after-parens, integer]
+        - include: not-whitespace-illegal
+    - match: '='
+      scope: keyword.operator.assignment.d
+      set: conditional-declaration-assignment
+    - match: '(?=\S)'
+      set: optional-block-statement-or-label
+
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.loop.d
+        2: keyword.control.loop.d
+      push: [optional-block-statement, foreach-arguments]
+
+  static-assert-in:
+    - match: '\b(static)\s+(assert)\b'
+      captures:
+        1: keyword.other.assert.d
+        2: keyword.other.assert.d
+      push: [expect-end-of-line, assert-arguments]
+
+  constructions-in:
+    - match: '{{construction_lookahead}}'
+      push: construction
+  construction:
+    # Post-blit constructor
+    - match: '\b(this)\s*(\()\s*(this)\s*(\))'
+      captures:
+        1: entity.name.function.post-blit.d
+        2: punctuation.section.group.begin.d
+        3: variable.language.d
+        4: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - match: '\b(this)\s*(?=\(|$)'
+      captures:
+        1: entity.name.function.constructor.d
+        2: punctuation.section.group.begin.d
+      set: constructor-argument-list
+    - match: '(~\s*this)\s*(\()\s*(\))'
       captures:
-        1: keyword.control.statement.d
+        1: entity.name.function.destructor.d
+        2: punctuation.section.group.begin.d
+        3: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - include: not-whitespace-illegal-pop
+  # This could either be function/template parameters or arguments. Assume
+  # arguments and switch to parameters when possible
+  constructor-argume
```

Comment:
You'll probably need to shuffle things around a little, but ideally the entirety of the labels and accessors should get a meta scope of `meta.path`.